### PR TITLE
Modify the snapshot API functions to expect a 32-bit aligned buffer pointer

### PR DESCRIPTION
**Pull request by robertsipka**
Fix for issue #1647

### CODE REVIEW

**Comment by zherczeg** [comment]
I really don't like these `Fix for issue` commit titles. Please write a real description.

File: jerry-core/ecma/base/ecma-literal-storage.c [hunk 1]
```diff
@@ -306,12 +307,14 @@ ecma_save_literals_for_snapshot (uint8_t *buffer_p, /**< [out] output snapshot b
    * constant so the first literal must have offset one. */
   uint32_t literal_offset = JERRY_SNAPSHOT_LITERAL_ALIGNMENT;
 
-  ((uint32_t *) buffer_p)[0] = string_count;
-  ((uint32_t *) buffer_p)[1] = number_count;
-  buffer_p += 2 * sizeof (uint32_t);
+  (buffer_p)[0] = string_count;
+  (buffer_p)[1] = number_count;
```
Comment (zherczeg) [inline jerry-core/ecma/base/ecma-literal-storage.c]: No need () around buffer_p anymore.

File: jerry-core/ecma/base/ecma-literal-storage.c [hunk 2]
```diff
@@ -478,9 +485,9 @@ ecma_load_literals_from_snapshot (const uint8_t *buffer_p, /**< buffer with lite
     return false;
   }
 
-  uint32_t string_count = ((uint32_t *) buffer_p)[0];
-  uint32_t number_count = ((uint32_t *) buffer_p)[1];
-  buffer_p += 2 * sizeof (uint32_t);
+  uint32_t string_count = (buffer_p)[0];
+  uint32_t number_count = (buffer_p)[1];
```
Comment (zherczeg) [inline jerry-core/ecma/base/ecma-literal-storage.c]: Ditto.

File: jerry-core/jerry-snapshot.c [hunk 3]
```diff
@@ -564,7 +566,7 @@ jerry_exec_snapshot (const void *snapshot_p, /**< snapshot */
 
   static const char * const invalid_version_error_p = "Invalid snapshot version";
   static const char * const invalid_format_error_p = "Invalid snapshot format";
-  const uint8_t *snapshot_data_p = (uint8_t *) snapshot_p;
+  const uint32_t *snapshot_data_p = snapshot_p;
```
Comment (zherczeg) [inline jerry-core/jerry-snapshot.c]: In this case I would keep the `const uint8_t *snapshot_data_p` form here. The asserts later are ok.

File: jerry-main/main-unix.c [hunk 4]
```diff
@@ -679,7 +679,7 @@ main (int argc,
                                                                 is_save_snapshot_mode_for_global_or_eval,
                                                                 false,
                                                                 snapshot_save_buffer,
-                                                                JERRY_BUFFER_SIZE);
+                                                                sizeof (snapshot_save_buffer));
```
Comment (zherczeg) [inline jerry-main/main-unix.c]: Isn't sizeof gives back the size in bytes? This is incorrect then.

Comment (robertsipka) [inline jerry-main/main-unix.c]: You're right!  I've updated the PR accordingly.

File: jerry-core/jerry-snapshot.c [hunk 5]
```diff
@@ -934,8 +937,10 @@ jerry_parse_and_save_literals (const jerry_char_t *source_p, /**< script source
     return 0;
   }
 
-  uint8_t * const buffer_start_p = buffer_p;
-  uint8_t * const buffer_end_p = buffer_p + buffer_size;
+  uint8_t *buffer_data_p = (uint8_t *) buffer_p;
+
+  uint8_t * const buffer_start_p = buffer_data_p;
+  uint8_t * const buffer_end_p = buffer_data_p + buffer_size;
```
Comment (zherczeg) [inline jerry-core/jerry-snapshot.c]: No space after *

File: jerry-core/jerry-snapshot.c [hunk 6]
```diff
@@ -968,43 +973,43 @@ jerry_parse_and_save_literals (const jerry_char_t *source_p, /**< script source
   if (is_c_format)
   {
     /* Save literal count. */
-    buffer_p = jerry_append_chars_to_buffer (buffer_p,
-                                             buffer_end_p,
-                                             (const char *) "jerry_length_t literal_count = ",
-                                             0);
+    buffer_data_p = jerry_append_chars_to_buffer (buffer_data_p,
```
Comment (zherczeg) [inline jerry-core/jerry-snapshot.c]: Buffer data sounds like data data, a word repeat to me. What about destination_p?

Comment (robertsipka) [inline jerry-core/jerry-snapshot.c]: Good idea, thanks! I've updated the patch.

File: tests/unit/test-api.c [hunk 7]
```diff
@@ -180,6 +185,8 @@ const jerry_char_ptr_t magic_string_items[] =
 #undef JERRY_MAGIC_STRING_DEF
 };
 
+
+
```
Comment (zherczeg) [inline tests/unit/test-api.c]: Remove these two new lines.

**Pull request merged**